### PR TITLE
T236 Update arlocal to 1.1.29 and integrate wallet with ArweaveProvider

### PR DESCRIPTION
**Pull request by decentdav**
- `key.json` is no longer required
  - Instead, we create a wallet and mint some fake AR tokens

### CODE REVIEW

File: src/providers/arweave-sync.jsx [hunk 1]
```diff
@@ -2,20 +2,27 @@ import React, { createContext, useContext, useState } from 'react';
 import PropTypes from 'prop-types';
 import { ToastContext } from './toast';
 import { createPodcast } from '../client';
+import { ArweaveContext } from './arweave';
 
 export const ArweaveSyncContext = createContext();
 
 function ArweaveSyncProvider({ children }) {
+  const { getWallet, getWalletAddress } = useContext(ArweaveContext);
```
Comment (decentdav) [inline src/providers/arweave-sync.jsx:10]: Not sure yet if this is working. ArSync is still too immature, but we're working on v0.5 & v1.0 :)

File: src/providers/arweave.jsx [hunk 2]
```diff
@@ -1,26 +1,46 @@
...
+          }
+          return walletAddress;
+        },
+
+        async getWallet() {
+          if (!wallet) {
+            await setWallet(arweave.createNewDevWallet());
+            await setWalletAddress(arweave.getWalletAddress(wallet));
+          }
+          return walletAddress;
```
Comment (decentdav) [inline src/providers/arweave.jsx:33]: this should `return wallet`

File: src/providers/arweave.jsx [hunk 3]
```diff
@@ -1,26 +1,46 @@
...
 
   return (
     <ArweaveContext.Provider
       value={{
         hasItemsToSync: !!(podcastsToSync.length || setPodcastsToSync.length),
 
+        async getWalletAddress() {
+          if (!walletAddress) {
+            if (!wallet) {
+              await setWallet(arweave.createNewDevWallet());
```
Comment (mmahdigh) [inline src/providers/arweave.jsx:21]: React setState functions are asynchronous _practically_  but they are not `async ` in a Javascript sense (i.e., they don't return a `promise`) so they can't be `awaited`. 

[Some context](https://stackoverflow.com/questions/54069253/usestate-set-method-not-reflecting-change-immediately).

If I correctly understand what you're trying to achieve here, I guess below code would be a good way to do it:
```
      async getWalletAddress() {
        if (walletAddress) return walletAddress;
        let devWallet = wallet;
        if (!wallet) {
          devWallet = await arweave.createNewDevWallet();
          setWallet(devWallet);
        }
        const walletAddr = await arweave.getWalletAddress(devWallet);
        setWalletAddress(walletAddr);
        return walletAddr;
      }
```

File: src/providers/arweave.jsx [hunk 4]
```diff
@@ -1,26 +1,46 @@
...
+            }
+            await setWalletAddress(arweave.getWalletAddress(wallet));
+          }
+          return walletAddress;
+        },
+
+        async getWallet() {
+          if (!wallet) {
+            await setWallet(arweave.createNewDevWallet());
```
Comment (mmahdigh) [inline src/providers/arweave.jsx:30]: Again this function should be rewritten because `awaiting` doesn't work on React state functions.

**Review by mmahdigh — CHANGES_REQUESTED** [review]

File: src/client/arweave/wallet.js [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+  mintDevTokens(wallet);
+  return wallet;
+}
+
+export async function getWalletAddress(wallet) {
+  return client.wallets.jwkToAddress(wallet);
+}
+
+async function mintDevTokens(walletAddress, numTokens = '100000000000000000') {
+  await client.api.get(`/mint/${walletAddress}/${numTokens}`);
```
[the remark this answers is not in the document]
Comment (decentdav) [reply src/client/arweave/wallet.js:14]: Got it. I intended to do the latter indeed.

File: src/providers/arweave.jsx (hunk 3 above)
Comment (decentdav) [reply src/providers/arweave.jsx:21]: Done

File: src/providers/arweave.jsx [hunk 6]
```diff
@@ -16,21 +16,24 @@ function ArweaveProvider({ children }) {
         hasItemsToSync: !!(podcastsToSync.length || setPodcastsToSync.length),
 
         async getWalletAddress() {
-          if (!walletAddress) {
-            if (!wallet) {
-              await setWallet(arweave.createNewDevWallet());
-            }
-            await setWalletAddress(arweave.getWalletAddress(wallet));
+          if (walletAddress) return walletAddress;
+          let devWallet = wallet;
```
Comment (decentdav) [inline src/providers/arweave.jsx:20]: Can we make this more DRY by calling `getWallet()`?

Comment (mmahdigh) [reply src/providers/arweave.jsx:20]: Yes that's a good idea. But you don't have access to `getWallet` in `getWalletAddress`. If you define both functions in the component's body, I reckon it'll be doable.

File: src/client/arweave/wallet.js [hunk 7]
```diff
@@ -2,7 +2,8 @@ import client from './client';
 
 export async function createNewDevWallet() {
   const wallet = await client.wallets.generate();
-  await mintDevTokens(wallet);
+  await mintDevTokens(await getWalletAddress(wallet));
+
```
Comment (decentdav) [inline src/client/arweave/wallet.js:6]: Accidental empty line. Fixed in final patch. (:

File: src/providers/arweave.jsx [hunk 8]
```diff
@@ -9,42 +9,42 @@ function ArweaveProvider({ children }) {
   const [episodesToSync, setEpisodesToSync] = useState([]);
```
Comment (decentdav) [inline src/providers/arweave.jsx:9]: I plan to remove `episodesToSync` in ArSync v0.5-1.0: only `podcastsToSync` should be used. The episodes to sync are better nested under `podcastsToSync.episodes`.

Then the `then()` banishment on line 42 shall bring much relief.

**Review by mmahdigh — APPROVED** [review]

**Pull request merged**